### PR TITLE
RPRBLND-1967: Improve "Sync process" for instances

### PR DESCRIPTION
**Pull request by DagerD**
### PURPOSE
Speed up sync process for final rendering

### EFFECT OF CHANGE
Sync process for final rendering became much faster

### TECHNICAL STEPS
- Implemented multi-threading for instanced objects
- Implemented USD instancing technique 
- Added const CHUNK_COUNT to define how many object per thread will be processed (I figured out that 500 is almost ideal for my PC, I think we need to investigate which is better for other PCs)
- Added is_rand argument to function get_temp_file. We need it because USD doesn't update material if name of mtlx file isn't changed
- Modified blender_data node to fit new sync logic

### NOTES FOR REVIEWERS
- https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderUSD/pull/512 is strictly needed for working with instancing.
- Here is a delay after syncing and before rendering which doesn't concern to sync process. The reason is HdRPR creates material for every instance (the same material).

### CODE REVIEW

File: src/hdusd/export/material.py [hunk 1]
```diff
@@ -89,32 +89,13 @@ def sync_update_all(root_prim, mat: bpy.types.Material):
         # removing rpr_materialx_node in all material_prims
         return None
 
-    surfacematerial = next(node for node in doc.getNodes()
-                           if node.getCategory() == 'surfacematerial')
+    surfacematerial = next(node for node in doc.getNodes() if node.getCategory() == 'surfacematerial')
 
     stage = root_prim.GetStage()
```
Comment (bnagirniak) [inline src/hdusd/export/material.py:94]: stage and surfacematerial values aren't used anymore, can be removed

File: src/hdusd/export/material.py [hunk 2]
```diff
@@ -89,32 +89,13 @@ def sync_update_all(root_prim, mat: bpy.types.Material):
...
-        sdf_path = mat_prim.GetPath().AppendChild('Materials').AppendChild(surfacematerial.getName())
-        sdf_path_old = next((target for target in rel_bind.GetTargets()), None)
-
-        # check if bind path is changed
-        if not sdf_path_old or sdf_path == sdf_path_old:
-            return None
-
-        usd_mat = UsdShade.Material.Define(stage, sdf_path)
-        bindings.UnbindAllBindings()
-        bindings.Bind(usd_mat)
```
Comment (bnagirniak) [inline src/hdusd/export/material.py:120]: Did you completely test this part, that your are going to remove it? Won't it break something else?

File: src/hdusd/export/mesh.py [hunk 3]
```diff
@@ -205,40 +205,8 @@ def sync(obj_prim, obj: bpy.types.Object, mesh: bpy.types.Mesh = None, **kwargs)
...
-                usd_material = UsdShade.Material.Get(stage, child.GetPath())
-                UsdShade.MaterialBindingAPI(usd_mesh).Bind(usd_material)
-                
-        return
-
-    original_prim = stage.GetPrimAtPath(f"/{sdf_name(obj.original)}")
-    if original_prim and original_prim.IsValid():
-        for child in original_prim.GetChildren():
-            if len(child.GetAuthoredPropertyNames()) > 0:
-                return
```
Comment (bnagirniak) [inline src/hdusd/export/mesh.py:238]: Did you completely test this part, that your are going to remove it? Won't it break something else?

**Review by bnagirniak — COMMENTED** [review]
There are some suspicious about code which is going to be removed. Would like to confirm that nothing won't be broken. Please test this.

File: src/hdusd/export/mesh.py (hunk 3 above)
Comment (DagerD) [reply src/hdusd/export/mesh.py:238]: Yes

File: src/hdusd/export/material.py (hunk 2 above)
Comment (DagerD) [reply src/hdusd/export/material.py:120]: Yes

**Review by bnagirniak — APPROVED** [review]

**Comment by bnagirniak** [comment]
Waiting for https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderUSD/pull/512 to be merged

**Comment by bnagirniak** [comment]
@DagerD please fix conflicts.
Waiting for #167 to be merged.

**Review by VascoPi — APPROVED** [review]
Looks good.

**Comment by bnagirniak** [comment]
#167 is merged, now we can merge this PR

**Pull request merged**